src/basic/process-util.c

```diff
@@ -275,21 +276,59 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
 }
 
 void rename_process(const char name[8]) {
+        static char *mm = NULL;
+        static size_t mm_size = 0;
+        size_t l;
+        char *old_mm = NULL;
+        size_t old_mm_size = 0;
+
+        /* This is a like a poor man's setproctitle(). It changes the comm field, argv[0], and also the glibc's
+         * internally used name of the process. For the first one a limit of 16 chars applies; to the second one in
+         * many cases one of 10 (i.e. length of "/sbin/init") — however if we have CAP_SYS_RESOURCES it is unbounded;
```

Comment:
well, `/proc/$PID/cmdline` when set up by the kernel consists of exactly the NUL-separated command line arguments, and nothing else. When we manipulate it however the way we used to, we only place one string at the beginning of it, and then set the rest to NUL. 

A client that reads this file could get confused by that, and read this as n command line arguments, with n > 1 and all but the first one are empty. Or to give an example:

This command line: `["foo", "bar", "bazz"]` is normally visible in `/proc/$PID/cmdline` as: `foo\0bar\0bazz\0`. Now, if we change our process title to "waldo" we so far turned it into `waldo\0\0\0\0\0\0\0\0`, which programs could rightfully parse as the command line `["waldo", "", "", "", "", "", ""]`. With the new patch we instead make this `waldo\0`, and that's much nicer and more correct...

Ultimately the difference doesn't matter too much, as we mostly just want to prettify "ps" output, and there the distinction doesn't matter, but we still should do the right thing if we can.